data/data/vsphere/pre-bootstrap/main.tf

```diff
@@ -17,6 +39,69 @@ provider "vsphereprivate" {
   allow_unverified_ssl = false
 }
 
+
+//// Zoning
+
+data "vsphere_datacenter" "datacenter_zoning" {
+  for_each = local.vcenter_region_zone_map
+  name     = each.value.datacenter
+}
+
+data "vsphere_compute_cluster" "cluster_zoning" {
+  for_each      = local.vcenter_region_zone_map
+  name          = each.value.cluster
+  datacenter_id = data.vsphere_datacenter.datacenter_zoning[each.key].id
+}
+
+data "vsphere_datastore" "datastore_zoning" {
+
+  for_each      = local.vcenter_region_zone_map
+  name          = each.value.datastore
+  datacenter_id = data.vsphere_datacenter.datacenter_zoning[each.key].id
+}
+
+data "vsphere_network" "network_zoning" {
+  for_each      = local.vcenter_region_zone_map
+  name          = each.value.network
+  datacenter_id = data.vsphere_datacenter.datacenter_zoning[each.key].id
+}
+
+resource "vsphere_tag_category" "region_tag_category" {
+  name        = "openshift-region"
+  description = "Added by openshift-install do not remove"
+  cardinality = "SINGLE"
+
+  associable_types = [
+    "Datacenter"
+  ]
+}
+
+resource "vsphere_tag_category" "zone_tag_category" {
+  name        = "openshift-zone"
+  description = "Added by openshift-install do not remove"
+  cardinality = "SINGLE"
+
+  associable_types = [
+    "Cluster"
+  ]
+}
+
+resource "vsphere_tag" "region_tags" {
+  for_each    = { for k, v in local.vcenter_region_zone_map : k => v.region }
```

Comment:
TODO: with the complex vcenters object flattened need to determine how to get the unique regions only.